### DISCUSSION TITLE
How to set up UART(1) ESP32-S3 for USB serial comms using CH343

### DISCUSSION THREAD

**Discussion by kdavies-kd** [micropython/micropython#15579] [ESP32] [answered]
Please can anyone help with correct setup and software.
I have the ESP32-S3 from pihut with the serial usb board CH343 set up on a breadboard.
ESP32-S3 setup with micropython.
Windows sees both devices, Serial ports on COM4 and REPL on COM5.
Arduino Lab for MicroPython to run apps, copy files and REPL on COM5
Termite on COM4, 9600,8N1 no handshake
Serial USB CH343 connected to D4(rx), & D5(tx)
with ESP32-S3 GPIO pins 8 for tx and 7 for rx
To set up UART I have used: 
>>> import machine
>>>uart = machine.UART(1, baudrate=9600, tx=8, rx=7, timeout=5000)
the next command uart
>>>uart
reports:
UART(1, baudrate=9600, bits=8, parity=None, stop=1, tx=8, rx=7, rts=-1, cts=-1, txbuf=256, rxbuf=256, timeout=500, timeout_char=0)

Command: uart.any()
>>>uart.any()
reports:
0
the info at by [Loboris](https://github.com/loboris/MicroPython_ESP32_psRAM_LoBo/wiki/uart) has helped but still not getting any results.

I have wrote a short serial test script that loops to uart.write('Hello from uart1') every second and on the CH343 bard can see the TXD light flash.

Can anyone shed some light on what I am doing wrong or missed?

**Comment by shariltumin** [comment, marked as answer]
I'm not sure how you connected your USB2SERIAL to your ESP32S3.
Since you're using tx=8, rx=7 on your ESP32S3, you should connect it like this:
  
```
    USB2SERIAL        ESP32S3
       TXD-----------Pin7 (rx=7)
       RXD-----------Pin8 (tx=8) 
       GND-----------GND
```

Basically, you just need to cross the tx/rx wires and connect the ground together.

**Reply by kdavies-kd** [reply]
Duh, so blindingly obvious, thank you. Working as expected.
Next step is to learn asyncio
